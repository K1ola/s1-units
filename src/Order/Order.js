import React from 'react';
import './Order.css';
import {getDate} from '../utils/getDate';

export default class Order extends React.PureComponent {
	getItems() {
		const {order: {items}} = this.props;

		if (!items || !items.length) {
			return null;
		}

		return items.map((item, index) => (
			<div
				key={index}
				className='Order-item'
			>
				{item}
			</div>
		))
	}

	render() {
		const {order: {shop, date}} = this.props;
		return (
			<div className='Order'>
				<div className='Order-header'>
					<span className='Order-shop'>{shop}</span>
					<span>{getDate(date)}</span>
				</div>

				<div className='Order-items'>
					{this.getItems()}
				</div>
			</div>
		)
	}
}
