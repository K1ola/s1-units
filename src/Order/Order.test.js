import React from 'react';
import Order from './Order';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import {fakeOrders} from '../mock/fakeOrders';

configure({ adapter: new Adapter() });

describe('Order component', () => {
//   it('render with empty state', () => {
//     const wrapper = shallow(<Order order={""}/>);
//     // wrapper.setProps({key: "", order: ""});
//     expect(toJson(wrapper)).toMatchSnapshot();
//   });

  it('render with fakecomponent', () => {
    const wrapper = shallow(<Order key={0} order={fakeOrders[0]}/>);
    // wrapper.setProps({key: 0, order: fakeOrders[0]});
    expect(toJson(wrapper)).toMatchSnapshot();
   });

   it('check empty values', () => {
    const wrapper = shallow(<Order order={fakeOrders[0]} />);
    wrapper.setProps({order: {}});
    expect(wrapper.find(".Order-items").children().length).toEqual(0);
  });
});
