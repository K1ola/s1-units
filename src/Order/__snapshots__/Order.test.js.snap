// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component check  1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component check 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component check empty values 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component render with default state 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component render with default state 2`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component render with default state: snapshot test 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order component render with empty state 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component render with empty state 2`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component render with empy state 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component render with empyy state 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component render with fake 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order component render with fakecomponent 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order component render with fakecomponet 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;
